csrc/ir/nodes.cpp

```diff
@@ -860,12 +860,18 @@ SqueezeOp::SqueezeOp(
       num_removed_broadcasts++;
       auto id = in_dom[i];
       TORCH_INTERNAL_ASSERT(
-          id->isBroadcast(), "Can not squeeze non-broadcasting dimension(s).");
+          id->isBroadcast() || id->isSymbolic(),
+          "Squeeze dimension should be either Symbolic or Broadcast. Found ",
+          id->getIterType());
       TORCH_INTERNAL_ASSERT(
           !id->hasExpandedExtent(), "Can not squeeze expanded dimension(s).");
```

Comment:
Can an ID have expanded extent which evaluates to runtime `1`?

Reply:
Aside from why you'd want to expand to size 1, I guess the answer should be yes.

Reply:
@zasdfgbnm Should we just remove this assertion, assuming the user input fusion is correctly set up?

Reply:
I don't know. Prior to introducing concretization, I think it's safe to do this check, assuming that the squeeze op is directly translated from our framework frontend. After introducing concretization, do we make it possible to generate expand of dynamic size zero?

Reply:
Hmm, when we concretize a fusion, if we see an expand to a size-1 domain, it should be a noop, right? Do we need to create an expanded domain in such a case? Can we leave it as just a normal (non-expanded) broadcast domain?

Reply:
OK, then we can leave this assertion.